### ISSUE TITLE
Enable Base64 decoding for googlesource.com

### ISSUE THREAD

**Issue by djkoloski** [mozilla/cargo-vet#394]
We host all of our code in gitiles, which doesn't provide a way to directly download individual files. You can however download files as base64 blobs by adding `?format=TEXT` to the URL. It would be great if we could specify this as an option in our import section, something like:

```toml
[imports.fuchsia]
url = "https://fuchsia.googlesource.com/supply-chain/audits.toml?format=TEXT"
decode = "base64"
```

Is this something that would be worth adding to cargo-vet?

**Comment by bholley** [comment]
As it happens, I ran into precise this issue earlier today trying to import the [ChromiumOS audits](https://chromium.googlesource.com/chromiumos/third_party/rust_crates/+/main/cargo-vet/audits.toml) into Firefox. Reading through the [relevant issue](https://github.com/google/gitiles/issues/7), I don't expect this to be resolved in gitiles anytime soon.

Given that this is likely a pretty unique need, I'd be inclined to just special-case it rather than adding a general flag. We could, for example, detect when we're fetching an import from googlesource.com, and if so, try base64-decoding it before parsing. If base64-decode fails, we'd just fall through to the regular parser in case the server behavior ever changes.

That said, given that multiple Google teams are now deploying cargo-vet, I wonder if we should just jump straight to deploying `cargo-vet aggregate`. That would spare external consumers from needing separate imports for ChromiumOS and Fuchsia (along with any other potential future deployments). The automation for that could handle the base64 encoding internally, and put the result in a dedicated location that was served as normal plaintext.

@djkoloski @gburgessiv thoughts?

**Comment by djkoloski** [comment]
> Reading through the https://github.com/google/gitiles/issues/7, I don't expect this to be resolved in gitiles anytime soon.

I read through that gitiles issue as well and agree that it looks... less than hopeful. I'll take a look and see if we can get any motion on it though, since it seems to be widely desirable. It's worth a shot at least.

> I'd be inclined to just special-case it rather than adding a general flag.

I think this could be a good solution in the immediate-term.

> I wonder if we should just jump straight to deploying `cargo-vet aggregate`.

I think we should do that. I'll follow up with some of our internal teams and see how we can approach resource allocation for it, since that's the cloudiest part. I anticipate that this could take some time though.

**Comment by afranchuk** [comment]
I really wish they'd modify the `content-type` header to indicate it's base64, but it seems [it was intentional](https://github.com/google/gitiles/blob/e54582fdd2a8783a6078f3a25eb08bb622f2e497/java/com/google/gitiles/PathServlet.java#L166) :(

**Comment by bholley** [comment]
Fixed by #397. I confirmed it works and submitted [some patches](https://bugzilla.mozilla.org/show_bug.cgi?id=1815773) to import the chromeos audits into firefox. 🎉
